controllers/oauth.go

```diff
@@ -36,10 +36,8 @@ func OAuth(w http.ResponseWriter, r *http.Request) {
 	db := app.Db
 
 	var reqFields = OAuthReqBodyFields{}
-	err := json.NewDecoder(r.Body).Decode(&reqFields)
-
+	err := utils.DecodeJSONBody(r, w, &reqFields)
 	if err != nil {
-		utils.LogErrAndRespond(r, w, err, "Error parsing JSON body parameters.", http.StatusBadRequest)
 		return
 	}
```

Comment:
Since the error has already been checked in the function, why a check here, also it loterally does nothing.  Either return error here instead of the funxtion or remove this check.